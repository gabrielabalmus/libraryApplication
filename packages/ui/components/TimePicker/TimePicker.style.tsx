import TextField, { TextFieldProps } from "@mui/material/TextField";
import { styled as MuiStyled } from "@mui/material/styles";

export const StandardTextField = MuiStyled(TextField)<TextFieldProps>(
  ({ theme }) =>
    theme.unstable_sx({
      marginBottom: "12px",
    })
);
