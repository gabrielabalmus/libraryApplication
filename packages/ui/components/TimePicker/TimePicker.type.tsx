import { Dayjs } from "dayjs";

export interface TimeProps {
  label: string;
  value: string;
  onChange: (value: Dayjs | null) => void;
  errorMessage?: string;
}
