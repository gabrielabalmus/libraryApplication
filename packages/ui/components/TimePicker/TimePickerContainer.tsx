import { TextFieldProps } from "@mui/material/TextField";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
import { TimePicker } from "@mui/x-date-pickers/TimePicker";
import { StandardTextField } from "./TimePicker.style";
import { TimeProps } from "./TimePicker.type";

const TimePickerContainer: React.FC<TimeProps> = ({
  label,
  value,
  errorMessage,
  onChange,
}) => {
  return (
    <LocalizationProvider dateAdapter={AdapterDayjs}>
      <TimePicker
        label={label}
        value={value}
        onChange={onChange}
        OpenPickerButtonProps={{ disableRipple: true }}
        renderInput={(params: TextFieldProps) => (
          <StandardTextField
            {...params}
            error={!!errorMessage}
            variant="standard"
            helperText={errorMessage}
          />
        )}
      />
    </LocalizationProvider>
  );
};

export default TimePickerContainer;
