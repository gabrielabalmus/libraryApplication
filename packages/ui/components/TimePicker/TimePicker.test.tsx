import { render } from "@testing-library/react";
import TimePicker from ".";
import { TimeProps } from "./TimePicker.type";
import "@testing-library/jest-dom";

const mockChange = jest.fn();

const renderTimePicker = ({ ...props }: TimeProps) =>
  render(<TimePicker {...props} />);

describe("snapshot tests for TimePicker", () => {
  it("renders correctly", () => {
    const { container } = renderTimePicker({
      label: "Test1",
      value: "2023-05-01T01:03:00+03:00",
      onChange: mockChange,
    });

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with error message", () => {
    const { container } = renderTimePicker({
      label: "Test1",
      value: "2023-05-01T01:03:00+03:00",
      onChange: mockChange,
      errorMessage: "Test error message",
    });

    expect(container).toMatchSnapshot();
  });
});
