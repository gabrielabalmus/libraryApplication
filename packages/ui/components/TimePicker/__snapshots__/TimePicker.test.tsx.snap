// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for TimePicker renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r0:"
      id=":r0:-label"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r0:"
        placeholder="hh:mm (a|p)m"
        type="tel"
        value="01:03 AM"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose time, selected time is 1:03 AM"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ClockIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot tests for TimePicker renders correctly with error message 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r1:"
      id=":r1:-label"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedEnd css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-describedby=":r1:-helper-text"
        aria-invalid="true"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r1:"
        placeholder="hh:mm (a|p)m"
        type="tel"
        value="01:03 AM"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose time, selected time is 1:03 AM"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ClockIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
      id=":r1:-helper-text"
    >
      Test error message
    </p>
  </div>
</div>
`;
